timeseries/src/autogluon/timeseries/models/chronos/model.py

```diff
@@ -254,7 +286,93 @@ def _fit(
-        self.time_limit = time_limit
+
+        fine_tune_time_limit = time_limit // 2 if self.fine_tune else 0
+        self.time_limit = time_limit - fine_tune_time_limit  # inference time budget
+
+        if self.fine_tune:
+            context_length = self._get_context_length(train_data)
+            # load model pipeline to device memory
+            self.load_model_pipeline(context_length=context_length, is_training=True)
+
+            output_dir = Path(self.path) / "transformers_logs"
+            training_args = TrainingArguments(
+                output_dir=str(output_dir),
+                per_device_train_batch_size=self.fine_tune_batch_size,
+                learning_rate=self.fine_tune_lr,
+                lr_scheduler_type="linear",
+                warmup_ratio=0.0,
+                optim="adamw_torch_fused",
+                logging_dir=str(output_dir),
+                logging_strategy="steps",
+                logging_steps=self.logging_steps,
+                save_strategy="steps",
+                save_steps=self.eval_and_save_steps,
+                evaluation_strategy="steps",
+                eval_steps=self.eval_and_save_steps,
+                report_to="none",
+                max_steps=self.fine_tune_steps,
+                gradient_accumulation_steps=1,
+                dataloader_num_workers=self.data_loader_num_workers,
+                tf32=self._has_tf32(),
+                save_only_model=True,
+                load_best_model_at_end=True,
+                metric_for_best_model="eval_loss",
+                prediction_loss_only=True,
+                save_total_limit=1,
+            )
+            tokenizer_train_dataset = ChronosFineTuningDataset(
+                tokenizer=self.model_pipeline.tokenizer,
+                target_df=train_data,
+                freq=self.freq,
+                target_column=self.target,
+                context_length=context_length,
+                prediction_length=self.prediction_length,
+                mode="training",
+            )
+            callbacks = [TimeLimitCallback(time_limit=fine_tune_time_limit), EvaluateAndSaveFinalStepCallback()]
+
+            tokenizer_val_dataset = (
+                ChronosFineTuningDataset(
+                    tokenizer=self.model_pipeline.tokenizer,
+                    target_df=val_data,
+                    freq=self.freq,
+                    target_column=self.target,
+                    context_length=context_length,
+                    prediction_length=self.prediction_length,
+                    mode="validation",
+                )
+                if val_data is not None
+                else None
+            )
+
+            trainer = Trainer(
+                model=self.model_pipeline.model.model,
+                args=training_args,
+                train_dataset=tokenizer_train_dataset,
+                eval_dataset=tokenizer_val_dataset,
+                callbacks=callbacks,
+            )
```

Comment:
Can we control the verbosity of the HF trainer based on the `verbosity` level provided to `_fit`?

Here is what the output currently looks like in a jupyter notebook:

![image](https://github.com/user-attachments/assets/8798d13f-9322-4652-90fb-43c43a2c1963)

We should try to completely suppress the HF logs + warnings for `verbosity<=2`.

If possible, it would be nice to show the progress bar + DF for `verbosity>=3` (but not critical).

Reply:
This is what I went with in the end:

```
# verbosity < 3: all logs and warnings from transformers will be supressed
# verbosity >= 3: progress bar and loss logs will be logged
# verbosity 4: everything will be logged
```